File: 1455/CPP/One.cpp
#include <iostream>
#include <vector>

using namespace std;

int isPrefixOfWord(string s, string w)
{
    int wordIndex = 1;
    int i = 0;

    while (i < s.length())
    {
        if (s.substr(i, w.length()) == w)
        {
            return wordIndex;
        }

        while (i < s.length() && s[i] != ' ')
            i++;
        while (i < s.length() && s[i] == ' ')
            i++;

        wordIndex++;
    }
    return -1;
}

int main()
{
    string s = "i love eating burger";
    string w = "burg";
    cout << isPrefixOfWord(s, w) << endl;
    return 0;
}

File: 1455/JavaScript/One.js
var isPrefixOfWord = function (sentence, searchWord) {
  let reg = new RegExp(`^${searchWord}`, "g");
  let match = sentence.split(" ").findIndex((w, i) => {
    if (reg.test(w)) return true;
    return false;
  });
  return match == -1 ? match : match + 1;
};

console.log(isPrefixOfWord("i love eating burger", "burg")); // 4
console.log(isPrefixOfWord("this problem is an easy problem", "pro")); // 2
